.circleci/config.yml

```diff
@@ -308,83 +300,66 @@ jobs:
 workflows:
   CI:
     jobs:
-      - checks:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/
+      - validate-docs
+      - checks
       - tests:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/
+          requires: 
+            - checks
+            - validate-docs
+
       - integration-tests:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/          
+          requires: 
+            - checks
+            - validate-docs
+
+      - tck-tests:
+          requires: 
+            - checks
+            - validate-docs
+
       - maven-java-tests:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/ 
+          requires: 
+            - checks
+            - validate-docs
+
       - samples-tests:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/ 
-      - tck-tests:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/
-      - validate-docs:
-          filters: # required since publish jobs have tag filters and dependencies
-            tags:
-              only: /.*/
+          requires: 
+            - checks
+            - validate-docs
+            
       - publish:
           filters: # version tags only
             tags:
               only: /^v.*/
             branches:
               ignore: /.*/
-          requires:
-            - checks
-            - tests
-            - integration-tests
-            - maven-java-tests
-            - samples-tests    
-            - tck-tests
-            - validate-docs   
+
       - publish-maven:
           filters: # version tags only
             tags:
               only: /^v.*/
             branches:
               ignore: /.*/
           requires:
-            - checks
-            - tests
-            - integration-tests
-            - maven-java-tests
-            - samples-tests    
-            - tck-tests
-            - validate-docs   
+            - publish
+
       - publish-tck:
           filters: # version tags only
             tags:
               only: /^v.*/
             branches:
               ignore: /.*/
           requires:
-            - checks
-            - tests
-            - integration-tests
-            - tck-tests
-            - validate-docs
+            - publish
```

Comment:
I'm wondering why the tck is an apart publishing step. Shouldn't that be just part of the main artifacts? 

Anyway, we can change on follow-up PR.

Reply:
Nevermind, I forgot that the TCK involves the publishing of a docker image. So, makes sense to have it's own step.